redbiom/fetch.py

```diff
@@ -376,14 +376,29 @@ def taxon_ancestors(context, ids, get=None, normalize=None):
         import redbiom
         config = redbiom.get_config()
         get = redbiom._requests.make_get(config)
+
+    hmgetter = redbiom._requests.buffered
+    remapped_bulk = hmgetter(iter(ids), None, 'HMGET', context,
+                             get=get, buffer_size=100,
+                             multikey='feature-index')
+
+    # map the feature identifier to an internal ID
+    # if an internal ID does not exist, keep the provided ID
+    # the provided ID is kept in the event a taxon name such as
+    # p__Firmicutes is provided
+    remapped = {name: id_ if id_ is not None else name
+                for names, idx in remapped_bulk
+                for name, id_ in zip(names, idx)}
+
     # bulk gather the taxonomy information for all the tips and their parents
-    to_get = ids
+    to_get = list(remapped.values())
```

Comment:
I'm actually surprised that we need a list here, do you know why? Is it cause it changes inside the while?

Reply:
Yes exactly, the algorithm walks up the hierarchy. Each call to `hmgetter` is effectively getting the parents of the present contents of `to_get`. This could also be a `set`, as it subsequently changes to within the loop in order to avoid fetching the same parent multiple times, doesn't really matter as long as it is mutable and iterable.